pyintegration/deephaven/appmode.py

```diff
@@ -0,0 +1,66 @@
+#
+#   Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+""" This module supports writing Deephaven application mode Python scripts. """
+from typing import Dict
+
+import jpy
+
+from deephaven import DHError
+from deephaven._wrapper import JObjectWrapper, wrap_j_object, unwrap
+
+_JApplicationContext = jpy.get_type("io.deephaven.appmode.ApplicationContext")
+_JApplicationState = jpy.get_type("io.deephaven.appmode.ApplicationState")
+
+
+class ApplicationState(JObjectWrapper):
+    """ The ApplicationState represents the state of an application. """
+    j_object_type = _JApplicationState
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_app_state
+
+    def __init__(self, j_app_state):
+        self.j_app_state = j_app_state
+
+    def __repr__(self):
+        return f"id: {self.j_app_state.id()}, name: {self.j_app_state.name()}"
+
+    def __getitem__(self, item):
+        item = str(item)
+        value = wrap_j_object(self.j_app_state.getField(item))
+        if not value:
+            raise KeyError(item)
+
+    def __setitem__(self, key, value):
+        key = str(key)
+        self.j_app_state.setField(key, unwrap(value))
```

Comment:
Note: the item you are returning from get is not the same item you are placing via set. I see you *are* calling `.value()` in fields() - I'm not sure if we want to have JObjectWrapper Field, or return `.value()` from a get.

Reply:
I think Python dict is a good fit, wrapping Field seems a bit overkill.

Reply:
Sure - but I just want to triple check:

```python
app = get_app_state()
app['hello'] = 'world'
print(app['hello'])
```
will *not* print `world` b/c getField returns a Field...

Reply:
Thanks for finding the bug!